src/System.Globalization.Extensions/tests/Normalization/NormalizationAll.cs

```diff
@@ -2,204 +2,127 @@
-                    failed |= CheckFailue(normForm, c4, strNFc2);
-                    failed |= CheckFailue(normForm, c4, strNFc3);
-                    failed |= CheckFailue(normForm, c4, strNFc4);
-                    failed |= CheckFailue(normForm, c4, strNFc5);
+                    AssertEqualsForm(c4, normalized1);
+                    AssertEqualsForm(c4, normalized2);
+                    AssertEqualsForm(c4, normalized3);
+                    AssertEqualsForm(c4, normalized4);
+                    AssertEqualsForm(c4, normalized5);
 
                     // c4 is normalized to Form KC
-                    if (!bNFc4)
-                    {
-                        Console.WriteLine("'{0}' is marked as not normalized with form {1}", c4, normForm);
-                        failed = true;
-                    }
+                    Assert.True(c4.IsNormalized(normForm), string.Format("'{0}' is marked as not normalized with form {1}", c4, normForm));
                     break;
 
                 case NormalizationForm.FormKD:
                     // c5 == NFKD(c1) == NFKD(c2) == NFKD(c3) == NFKD(c4) 
                     //    == NFKD(c5)
-                    failed |= CheckFailue(normForm, c5, strNFc1);
-                    failed |= CheckFailue(normForm, c5, strNFc2);
-                    failed |= CheckFailue(normForm, c5, strNFc3);
-                    failed |= CheckFailue(normForm, c5, strNFc4);
-                    failed |= CheckFailue(normForm, c5, strNFc5);
+                    AssertEqualsForm(c5, normalized1);
+                    AssertEqualsForm(c5, normalized2);
+                    AssertEqualsForm(c5, normalized3);
+                    AssertEqualsForm(c5, normalized4);
+                    AssertEqualsForm(c5, normalized5);
 
                     // c5 is normalized to Form KD
-                    if (!bNFc5)
-                    {
-                        Console.WriteLine("'{0}' is marked as not normalized with form {1}", c5, normForm);
-                        failed = true;
-                    }
+                    Assert.True(c5.IsNormalized(normForm), string.Format("'{0}' is marked as not normalized with form {1}", c5, normForm));
                     break;
             }
-            return failed;
         }
 
-        private static bool CheckFailue(NormalizationForm normForm, string c, string cForm)
+        private static void AssertEqualsForm(string c, string cForm)
         {
-            if (!c.Equals(cForm))
-            {
-                Console.WriteLine("'{0}' is not matched with the normalized form '{1} with {2} normalization", DumpStringAsCodepoints(c), DumpStringAsCodepoints(cForm), cForm);
-                return true;
-            }
-
-            return false;
+            Assert.True(c.Equals(cForm), string.Format("'{0}' is not matched with the normalized form '{1} with {2} normalization", DumpStringAsCodepoints(c), DumpStringAsCodepoints(cForm), cForm));
```

Comment:
Nit: I prefer using $"{DumpStringAsCodepoints(c) }...." instead of using String.Format here. just for the code readability.